src/vaults/VaultManager.ts

```diff
@@ -711,6 +714,7 @@ class VaultManager {
 
     if ((await this.getVaultName(vaultId, tran)) == null) return;
     await this.vaultLocks.withF([vaultId, RWLockWriter, 'write'], async () => {
+      await tran.lock(vaultId);
```

Comment:
This needs to use the a unique key path relative to the vault manager.

I recommend using the same key path and just joining it to a `string`.

Otherwise this can result in global namespace conflict.

This goes for any usage of `tran.lock`.

I think you're doing something similar already with:

```
      await tran.lock(
        [...this.vaultsNamesDbPath, newVaultName].toString(),
        vaultId,
      );
```

It would rely on the array `toString()`. I reckon instead using `[].join('')` cause it will ensure deterministic result rather than relying on the magic `toString()`.

Reply:
Actually maybe we could just take the key path. I just checked:

```
> [Buffer.from('abc'), 'lol', 123].toString()
'abc,lol,123'
```

This seems to work well enough. Let's hope the `toString` of arrays don't change in the mean time.

Reply:
I think this is more deterministic and better:

```
> [Buffer.from('abc'), 'lol', 123].join('')
'abclol123'
```

So just pass the same key path and and call a `join('')`.

Reply:
Resolved. I also updated all usages of locking on a KeyPath to use `join` instead of `toString`